src/mlpack/tests/adaboost_test.cpp

```diff
@@ -671,12 +692,32 @@ BOOST_AUTO_TEST_CASE(ClassifyTest_NONLINSEP)
   AdaBoost<DecisionStump<> > a(inputData, labels.row(0), numClasses, ds,
       iterations, tolerance);
 
-  arma::Row<size_t> predictedLabels(testData.n_cols);
-  a.Classify(testData, predictedLabels);
+  arma::Row<size_t> predictedLabels1(testData.n_cols),
+                    predictedLabels2(testData.n_cols);
+  arma::mat probabilities;
+
+  a.Classify(testData, predictedLabels1);
+  a.Classify(testData, predictedLabels2, probabilities);
+
+  BOOST_REQUIRE_EQUAL(probabilities.n_cols, testData.n_cols);
+
+  for (size_t i = 0; i < predictedLabels1.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(predictedLabels1[i], predictedLabels2[i]);
+
+  arma::colvec pRow;
+  arma::uword maxIndex = 0;
+
+  for (size_t i = 0; i < predictedLabels1.n_cols; i++)
+  {
+    pRow = probabilities.unsafe_col(i);
+    pRow.max(maxIndex);
+    BOOST_REQUIRE_EQUAL(predictedLabels1(i), maxIndex);
+    BOOST_REQUIRE_EQUAL(arma::accu(probabilities.col(i)), 1);
```

Comment:
This should be `BOOST_REQUIRE_CLOSE()` also; I can fix it during merge, or you can do it beforehand, either way is fine.